coalib/bears/Bear.py

```diff
@@ -274,6 +293,27 @@ def log_message(self, log_message, timestamp=None, **kwargs):
     def run(self, *args, dependency_results=None, **kwargs):
         raise NotImplementedError
 
+    def debugger_config(self,*args,**kwargs):
+        if (isinstance(self.run(*args, **kwargs), types.GeneratorType)) and \
+            ((u'wrapping_function of' in str(self.run)) or
+                (u'_new_func of' in str(self.run))):
+            kwargs['debug_flag'] = self.debug_flag
+            debug_mode_function(self.run,*args,**kwargs)
+            kwargs.pop('debug_flag')
+
+        elif (isinstance(self.run(*args, **kwargs), types.GeneratorType)):
+            dbg.runcall(self.run,*args, **kwargs)
+            debug_mode_function(self.run,*args,**kwargs)
+
+        elif (isinstance(self.run(*args, **kwargs),(str,type(None)))) and \
```

Comment:
You are rerunning `run` all the time, this is not okay. Run it once, and then check the return value.

Reply:
Hm you want to get the return type by running it... this is not a good design. The problem is we can't predict the type without running the bear, so you have to find something different.